pkg/util/cloudproviders/cloudproviders.go

```diff
@@ -147,3 +147,23 @@ func GetPublicIPv4(ctx context.Context) (string, error) {
 	log.Infof("No public IPv4 address found")
 	return "", errors.New("No public IPv4 address found")
 }
+
+type cloudProviderAccountIDCollector struct {
+	name         string
+	metadataName inventories.AgentMetadataName
+	callback     func(context.Context) (string, error)
+}
+
+func CollectAccountID(ctx context.Context) {
```

Comment:
I think this could be done automatically in `DetectCloudProvider`. If we detect one, we try to fetch its account ID.